src/python/pants/backend/jvm/subsystems/zinc.py

```diff
@@ -49,13 +49,17 @@ def register_options(cls, register):
           Shader.exclude_package('scala', recursive=True),
           Shader.exclude_package('xsbt', recursive=True),
           Shader.exclude_package('xsbti', recursive=True),
+          # Unfortunately, is loaded reflectively by the compiler.
+          Shader.exclude_package('org.apache.logging.log4j', recursive=True),
```

Comment:
Does that mean that when compiling sources that depend on a different version of log4j, there could be collision or binary incompatibility?

Reply:
Yes. Unavoidable, as far as I can tell. I confirmed that zinc fails to run without this exception.

Reply:
I'm fine shipping this, but maybe worth sandboxing this internally first and see how much pain it might cause if any?

Reply:
@wisechengyi : This only affects annotation processors, of which we have relatively few. I've tested it with them.